progressist/__init__.py

```diff
@@ -115,12 +115,28 @@ def speed(self):
         """Number of iterations per second."""
         return Float(1.0 / self.avg if self.avg else 0)
 
+    @property
+    def throttled(self):
+        if not self.throttle:
+            return
+        if isinstance(self.throttle, (int, float)):
+            throttle = self.throttle
+            if isinstance(self.throttle, float):
+                throttle = max(1, self.total * self.throttle)
+            throttle = self._last_render + throttle
+            if self.done < throttle:
+                if not self.total or throttle <= self.total:
```

Comment:
> If we haven't reached the total, then you should not print, because if the throttle is higher than the total, then the next time you print is when you reach the total.

Wow, very nice catch! :)